### PR TITLE
Add new errors: unknown params key, invalid escaped character, type mismatch

### PR DESCRIPTION
**Pull request by jsdidierlaurent**
- [x] Core
- [x] UI

### CODE REVIEW

File: monitorable/config/models/config.go [hunk 1]
```diff
@@ -12,23 +12,23 @@ type (
...
-		Tiles           []Tile `json:"tiles,omitempty"`
-		URL             string `json:"url,omitempty"`
-		InitialMaxDelay *int   `json:"initialMaxDelay,omitempty"`
+		Tiles           []ConfigTile `json:"tiles,omitempty"`
+		URL             string       `json:"url,omitempty"`
+		InitialMaxDelay *int         `json:"initialMaxDelay,omitempty"`
 
-		// Used by config.hydrate only (will be removed before returning config to UI)
-		ConfigVariant string `json:"configVariant,omitempty"`
+		// Used to validate config and for create api url (will be removed before returning config to UI)
```
Comment (Alex-D) [inline monitorable/config/models/config.go]: for => to

Comment (Alex-D) [reply monitorable/config/models/config.go]: // Used to validate config and to create API URLs
// Will be removed before being returned to the UI

File: monitorable/config/models/config.go [hunk 2]
```diff
@@ -53,6 +53,7 @@ const (
 	ConfigErrorInvalidFieldValue       ConfigErrorID = "ERROR_INVALID_FIELD_VALUE"
 	ConfigErrorMissingRequiredField    ConfigErrorID = "ERROR_MISSING_REQUIRED_FIELD"
 	ConfigErrorUnauthorizedField       ConfigErrorID = "ERROR_UNAUTHORIZED_FIELD"
+	ConfigErrorUnknownField            ConfigErrorID = "ERROR_UNKNOWN_FIELD"
```
Comment (Alex-D) [inline monitorable/config/models/config.go]: Keep things alphabetized :)

File: monitorable/config/models/config_parsing.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package models
+
+import (
+	"bytes"
+	"encoding/json"
+)
+
+// The goal here is to raise an error if a key is sent that is not supported.
+// This should stop many problems, like misspelling a parameter.
+type ConfigWrapper Config // Use wrapper to avoid infinite UnmarshalJSON loop
```
Comment (Alex-D) [inline monitorable/config/models/config_parsing.go]: Hum, that's not a wrapper, more like a TempConfig or something like that.

File: monitorable/config/models/config.go [hunk 4]
```diff
@@ -12,23 +12,23 @@ type (
...
+		Type models.TileType `json:"type"`
 
 		Label      string `json:"label,omitempty"`
 		RowSpan    *int   `json:"rowSpan,omitempty"`
 		ColumnSpan *int   `json:"columnSpan,omitempty"`
 
-		Tiles           []Tile `json:"tiles,omitempty"`
-		URL             string `json:"url,omitempty"`
-		InitialMaxDelay *int   `json:"initialMaxDelay,omitempty"`
+		Tiles           []ConfigTile `json:"tiles,omitempty"`
```
Comment (Alex-D) [inline monitorable/config/models/config.go]: `TileConfig`

File: monitorable/config/usecase/getconfig.go [hunk 5]
```diff
@@ -56,12 +66,62 @@ func (cu *configUsecase) GetConfig(params *models.ConfigParams) *models.ConfigBa
 				field = subMatch[0][1]
 			}
 
+			var configField = structs.Fields(models.Config{})
```
Comment (jsdidierlaurent) [inline monitorable/config/usecase/getconfig.go]: To be consistent with the rest of the code, replace the var xxx = aaa by xxx := aaa

**Pull request merged**